cmd/gcp-controller-manager/node_csr_approver_test.go

```diff
@@ -398,9 +398,25 @@ func TestValidators(t *testing.T) {
 			b.ips = []net.IP{net.ParseIP("1.2.3.4")}
 			b.dns = []string{"i0.z0.c.p0.internal", "i0.c.p0.internal", "i0"}
 		}
+		dualStackCase := func(b *csrBuilder, c *controllerContext) {
+			c.gcpCfg.ProjectID = "p0"
+			c.gcpCfg.Zones = []string{"z1", "z0"}
+			b.requestor = "system:node:ds0"
+			b.ips = []net.IP{net.ParseIP("1.2.3.4"), net.ParseIP("fd20:9b0:892f:800:0:5::")}
+			b.dns = []string{"ds0.z0.c.p0.internal", "ds0.c.p0.internal", "ds0"}
+		}
+		dualStackExtCase := func(b *csrBuilder, c *controllerContext) {
```

Comment:
whats the difference between dualStackCase and dualStackExtCase?

Reply:
There are 2 types IPv6 access types on subnets: internal and external. This is akin to public and private IPs for IPv4. However, in IPv6 case a VM can only have one type of IP address. When the subnet is 'internal', the instance IPv6 addresses are in the networkInterface.Ipv6Address field. When the instance is in an 'external' subnet, the IP address is populated under Ipv6AccessConfigs.ExternalIpv6 field.

Reply:
gotcha! thanks!